.buildkite/commands/command-that-calls-other-ps1-failing.ps1

```diff
@@ -0,0 +1,14 @@
+# Alone, this does not fail
+# & "$PSScriptRoot\command-that-fails-internally.ps1"
+
+# To make the script fail, we need to check the error
+& "$PSScriptRoot\command-that-fails-internally.ps1"
+
+if ($LASTEXITCODE -ne 0) {
+    # Using Write-Error doesn't print the exit code...
+    # Write-Error "Script failed with exit code $LASTEXITCODE"
+    #
+    # ...will using Write-Host be better?
+    Write-Host "Script failed with exit code $LASTEXITCODE"
+    exit $LASTEXITCODE
+}
```

Comment:
This is the experiment from which I derived the recommendation to **always follow an external command or script invocation with an exit code check.**

The check is verbose here because of the investigation, but we can do it in compact form like:

```ps1
if ($LastExitCode -ne 0) { exit $LastExitCode }
```

Also notice that by default PS1 is case-insenstive: So `$LASTEXITCODE`, `$LastExitCode`, and `$lastexitcode` are equivalent. My understanding is that `$LastExitCode` is the preferred form.